File: TsinghuaNet.Helper/Helpers/FluxHelper.vb
Public Module FluxHelper
    Public ReadOnly BaseFlux As ByteSize = ByteSize.FromGigaBytes(25)

    Public Function GetMaxFlux(flux As ByteSize, balance As Decimal) As ByteSize
        Return New ByteSize(Math.Max(flux.Bytes, BaseFlux.Bytes)) + ByteSize.FromGigaBytes(balance / 2)
    End Function
End Module

File: TsinghuaNet/FluxHelper.cs
using System;
using TsinghuaNet.Models;

namespace TsinghuaNet
{
    public static class FluxHelper
    {
        public static readonly ByteSize BaseFlux = ByteSize.FromGigaBytes(25);

        public static ByteSize GetMaxFlux(ByteSize flux, decimal balance)
        => ByteSize.FromBytes(Math.Max(flux.Bytes, BaseFlux.Bytes)) + ByteSize.FromGigaBytes((double)balance / 2);
    }
}
